.gitlab-ci.yml

```diff
@@ -31,8 +31,6 @@ stages:
 .branch-rules-main:
   rules:
     - if: $CI_DEPLOY_FREEZE == null && $CI_COMMIT_BRANCH == "main"
-      when: on_success
```

Comment:
Removing this would cause the deploy job to schedule/run even if the jobs in earlier stages will fail. This would also make the deployment to fail even though job would succeed. This was the safeguard to prevent deployment failure. 
Refer gitlab docs here https://docs.gitlab.com/ee/ci/yaml/#when

Reply:
@kamaldodiya My issue was that live deployments deployed automatically instead of waiting for manual deployment I checked another project and found they don't have these two lines.

Reply:
> My issue was that live deployments deployed automatically instead of waiting for manual deployment

Reply:
is adding  `when: on_success` enough?

Reply:
Couldn't find it in the docs, but [it seems that](https://stackoverflow.com/questions/75852822/gitlab-ci-is-there-a-difference-between-a-rule-with-whenalways-to-a-rule-with) `rules:if:when:` overrides `when:` rule. Therefore setting `rules:if:when: manual` (and removing `when: manual` from the jobs) should work.

Reply:
@kamaldodiya adding `when: on_success` should not be needed ([default value](https://docs.gitlab.com/ee/ci/yaml/#ruleswhen)).